### PR TITLE
Sushi UST-WETH strategy test

### PR DESCRIPTION
**Pull request by JasperSlimmens**
I wrote a test for a strategy farming the UST-WETH LP on Sushi.

I used other tests that I found in the folder and combined some aspects. The test will have a user supply LP to sushiswap. It then deposits LP tokens into harvest. The strategy will be simulated for 40 hours, 240 blocks per hour with a doHardWork() call every hour. It will show share price increase every hour. After 40 hours the user will withdraw, after which start and final balance for the user are shown.

The test can be run by using this command in one terminal to run a Mainnet Fork:

npx ganache-cli --fork https://eth-mainnet.alchemyapi.io/v2/YOUR_KEY_HERE -u '0xf00dD244228F51547f0563e60bCa65a30FBF5f7f' -u '0x0F4ee9631f4be0a63756515141281A3E2B293Bbe' -u '0xD6216fC19DB775Df9774a6E33526131dA7D19a2c'

The unlocked adresses are the Harvest Deployer, and whales for UST and WETH.

When the fork is up run the following command in another terminal:

truffle build && set MAINNET_FORK=True && npx truffle test ./test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js

### CODE REVIEW

File: test/mainnet-fork-test-config.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ module.exports = {
   DELAY_MINTER_ADDRESS: "0x284D7200a0Dabb05ee6De698da10d00df164f61d",
   FEE_REWARD_FORWARDER_ADDRESS: "0xEF08A639cAc2009fdAD3773CC9F56D6a8feB1153",
 
-  DAI_WHALE_ADDRESS: "0x6DCb8492B5De636fD9e0a32413514647D00eF8D0",
+  DAI_WHALE_ADDRESS: "0xa3692f4c74E36b984551F329239E7387eEF63f26",
```
Comment (belbix) [inline test/mainnet-fork-test-config.js:21]: why you don't use exciting whale addresses?
without this address, other tests will not work

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+          feeRewardForwarder = await FeeRewardForwarder.new(storage.address, underlying.address, MFC.UNISWAP_V2_ROUTER02_ADDRESS, { from: governance });
+          // set up controller
+          controller = await Controller.new(storage.address, feeRewardForwarder.address, {
+            from: governance,
+          });
+
+          await storage.setController(controller.address, { from: governance });
+
+          // set up the vault with 100% investment
+          vault = await makeVault(storage.address, underlying.address, 100, 100, {from: governance});
```
Comment (belbix) [inline test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:127]: why the vault has 100% investment?

File: test/mainnet-fork-test-config.js (hunk 1 above)
Comment (JasperSlimmens) [reply test/mainnet-fork-test-config.js:21]: I found the whales were outdated and were no longer whales for some of the tokens. I only changed the ones where I ran into trouble when trying to run tests. Not all are needed for the test I provided. I could change them back.

But as you can see here: https://etherscan.io/address/0x6DCb8492B5De636fD9e0a32413514647D00eF8D0. The old DAI whale does not have any DAI.

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js (hunk 2 above)
Comment (JasperSlimmens) [reply test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:127]: That is part of the original Sushi test file. Did not change the numbers. Was assuming this is the standard value.

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+          for (let i = 0; i < hours; i++) {
+            console.log("loop ", i);
+            let blocksPerHour = 240;
+            oldSharePrice = new BigNumber(await vault.getPricePerFullShare());
+            await vault.doHardWork({from: governance});
+            newSharePrice = new BigNumber(await vault.getPricePerFullShare());
+
+            console.log("old shareprice: ", oldSharePrice.toFixed());
+            console.log("new shareprice: ", newSharePrice.toFixed());
+            console.log("growth: ", (newSharePrice.dividedBy(oldSharePrice)).toFixed());
```
Comment (belbix) [inline test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:208]: the test should have assertions

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+          console.log("new farmer balance: ", farmerNewBalance.toFixed());
+          console.log("growth: ", (farmerNewBalance.dividedBy(farmerOldBalance)).toFixed());
+          Utils.assertBNGt(farmerNewBalance, farmerOldBalance);
+        });
+      });
+    }
+
+    // Can only uncomment one at a time because of the overlap in WETH
+
+    // test("WETH_USDT",
```
Comment (belbix) [inline test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:224]: create logic for it or remove

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js (hunk 2 above)
Comment (JasperSlimmens) [reply test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:127]: Also there is no reason to not invest 100%, as withdrawal is always possible still.

File: test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js (hunk 3 above)
Comment (JasperSlimmens) [reply test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:213]: Added assertions.

Comment (JasperSlimmens) [reply test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js:213]: Added assertions.

File: package.json [hunk 5]
```diff
@@ -20,6 +20,8 @@
   },
   "dependencies": {
     "@openzeppelin/contracts": "^2.5.0",
+    "@openzeppelin/contracts-ethereum-package": "^2.5.0",
+    "@openzeppelin/upgrades": "^2.8.0",
```
Comment (JasperSlimmens) [inline package.json:24]: These two packages I had to install as described here: https://www.notion.so/How-To-test-2215ca8fb6354f8183d1ea70a65f9f2d. Not sure if it is necessary to push this change in package.json. I can remove it.

File: test/mainnet-fork-test-config.js [hunk 6]
```diff
@@ -42,6 +42,7 @@ module.exports = {
   // WETH
   WETH_ADDRESS: "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
   WETH_WHALE_ADDRESS: "0x767ecb395def19ab8d1b2fcc89b3ddfbed28fd6b",
+  WETH_WHALE_ADDRESS_NEW: "0x0F4ee9631f4be0a63756515141281A3E2B293Bbe",
```
Comment (JasperSlimmens) [inline test/mainnet-fork-test-config.js:45]: Old WETH whale is no longer holding WETH.

**Comment by breadforthepeople** [comment]
Thank you for your contribution!
We've just released a new repo that makes things a lot easier: https://github.com/harvest-finance/harvest-strategy (including the issue with the changing whale). You are invited to contribute to this one instead. It will be cleaner, more sustainable, way faster, and easier for us to deploy.

**Pull request closed without merging**